### PR TITLE
Allow users to change the server description

### LINKED ISSUE #4412: Allow non-admins to change server description
### Is there an existing feature request for this?

- [X] I have searched the existing issues before opening this feature request.

### Describe the feature you would like to see.

Currently server owners and subusers with the `settings.rename` permission can only change the server name, not the description.

### Describe the solution you'd like.

Grant server owners the ability to change the server description by default. It would also be useful to introduce a new permission separate from `settings.rename` so that subusers can also change the server description, in parallel with the `settings.rename` permission for subusers.

### Additional context to this request.

N/A

### PR DESCRIPTION
**Pull request by Boy132**
Closes #4412.

This allows users (with the permission `settings.rename`) to change the server desc, similar to the server name.

image

### CODE REVIEW

**Comment by devnote-dev** [comment]
IMO "change-description" doesn't really fit for the permission name or endpoint... perhaps just "description" instead, to keep things concise?

**Comment by Boy132** [comment]
> IMO "change-description" doesn't really fit for the permission name or endpoint... perhaps just "description" instead, to keep things concise?

You're right, I changed it now.
For some reason my brain thought the endpoint for the docker image is called "change docker image", so it felt right with the "change". :D

**Comment by DaneEveritt** [comment]
Maybe just use the name permission? Seems weird to allow a user to only change one or the other, kinda unnecessarily granular IMO.

Also, description should probably be a textarea, not a text field?

**Comment by Boy132** [comment]
> Maybe just use the name permission? Seems weird to allow a user to only change one or the other, kinda unnecessarily granular IMO.

I'm fine with both variants. I guess it's up to Matthew.

> Also, description should probably be a textarea, not a text field?

Since line breaks aren't displayed, I didn't see a reason for a text area:
image

**Comment by DaneEveritt** [comment]
I was just thinking textarea more for readability — a long description is going to be annoying to edit when you have to scroll left/right IMO.

**Comment by richiedevs** [comment]
I agree with Dane.

**Comment by joshcannotcode** [comment]
I think it would be better if the name of the "Change Server Name" box was changed to "Change Server Details" and the description input is added to the "Change Server Name" box rather than being in a separate one, like this:
image

**Comment by devnote-dev** [comment]
@notjoshmiles They're separate because they use 2 different endpoints, and I don't know if the form is capable of submitting 2 requests at the same time. If we were to go with Dane's suggestion though, then I agree that it would be better like this.

**Comment by matthewpi** [comment]
If the endpoints are separate, then we should modify one to support updating both.  There should be one box for both these fields, not two individual ones.

**Comment by Boy132** [comment]
> If the endpoints are separate, then we should modify one to support updating both. There should be one box for both these fields, not two individual ones.

Made some changes: it now uses the "rename" endpoint (and rename permission) to change both the server name and/ or description. Also, the input for the description is now a textarea.
image

File: app/Http/Controllers/Api/Client/Servers/SettingsController.php [hunk 1]
```diff
@@ -34,8 +34,10 @@ public function __construct(
      */
     public function rename(RenameServerRequest $request, Server $server): JsonResponse
     {
+        $description = $request->input('description');
         $this->repository->update($server->id, [
             'name' => $request->input('name'),
+            'description' => $description !== null ? $description : '',
```
Comment (matthewpi) [inline app/Http/Controllers/Api/Client/Servers/SettingsController.php]: `$request->input('description') ?? ''`

**Review by matthewpi — CHANGES_REQUESTED** [review]

**Review by matthewpi — APPROVED** [review]

**Pull request merged**